iOS/UI/Reader/ReaderViewController.swift

```diff
@@ -382,17 +382,28 @@ extension ReaderViewController: ReaderHoldingDelegate {
         else {
             return nil
         }
+
         let skipDuplicates = UserDefaults.standard.bool(forKey: "Reader.skipDuplicateChapters")
-        let markSkipped = UserDefaults.standard.bool(forKey: "Reader.markSkippedChapters")
+        let markDuplicates = UserDefaults.standard.bool(forKey: "Reader.markDuplicateChapters")
+
         index -= 1
+        var nextChapterInList: Chapter?
+
         while index >= 0 {
             let new = chapterList[index]
-            if !skipDuplicates || (new.chapterNum != chapter.chapterNum || new.volumeNum != chapter.volumeNum) {
-                return new
+            let isDuplicate = new.chapterNum == chapter.chapterNum && new.volumeNum == chapter.volumeNum
```

Comment:
i think this will mark chapters that have no chapter or volume number as duplicates, but they probably shouldn’t be